### DISCUSSION TITLE
Assets that pay interest daily

### DISCUSSION THREAD

**Discussion by SW4T400** [edtechre/pybroker#133] [Q&A] [answered]
Hi,

I intend to work with crypto staking were one provides some coins and in return is rewarded some interest on the provided coins.  In the case of https://www.binance.com/en/earn/simple-earn they pay out interest **daily** to your spot account.

For the backtest this would just be a regular trade, but an interest is payed out daily to my cash balance as long as i hold some amount of this asset (position).

What are your thoughts on implementing this kind of interest payout. 
Would the best place to start be a sort of "inverse cost"? (I think I saw something in the docs about a custom cost model? Or was it slippage?)

Can this be implemented by the user at this point or would that require changes? If so - where would I start?

Any comment would be appreciated.

**Comment by edtechre** [comment, marked as answer]
Hi @SW4T400,

There is a [SlippageModel](https://www.pybroker.com/en/latest/reference/pybroker.slippage.html) but I don't think that will apply here. 

This would require changes on your end. What you could do is add the earned interest to the cash field in the Portfolio:
https://www.pybroker.com/en/latest/reference/pybroker.portfolio.html#pybroker.portfolio.Portfolio.cash

The ExecContext has a "private" reference to the Portfolio (_portfolio).

**Reply by SW4T400** [reply]
Ok. So I would look at my positions and some indicator or column that contains the interest rates to calculate the amount I would get and use this private reference to increase cash.

Since I want to do this every day on my 1d bars (as long as I hold any position) I would to this in fn of strategy.add_excecution OR in the case of your [newest example](https://medium.com/@edtechre/portfolio-optimization-with-pybroker-5ce0af389bfa):

```
def rebalance(ctxs: dict[str, ExecContext]):
   ------> if i have any positions => add interest to cash
    if start_of_month(ctxs):
        target = 1 / len(ctxs)  # Equal weighting
        set_target_shares(ctxs, {symbol: target for symbol in ctxs.keys()})
```

Am I correct to assume that set_after_exec is also done on every bar in the backtest? Just like fn in strategy.add_execution?

**Reply by edtechre** [reply]
Yes, set_after_exec is executed on every bar.

**Reply by Djangu-algo** [reply]
Ed, could you please add an example of this (interest added to cash balance) into either docs or just here?

**Comment by SW4T400** [comment]
@edtechre  I have a follow up question. If the interest/reward is in the form of shares/coins I would have to adjust the number of shares held - is this a value I can modify ? Instead of portfolio cash can I somehow bump my fractional shares to some higher value?

Or would I have to place a buy order with current prices and NO FEES to simulate this? What is your recommendation?

**Reply by edtechre** [reply]
@SW4T400 Sorry, I missed this follow-up. Were you able to figure this out?

**Reply by Heufi** [reply]
No worries. I made a fork and let Claude do it^^
